src/integration/transaction/application/services/transaction.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { randomUUID } from 'crypto';
+import { RawTxDto } from 'src/blockchain/ain/jellyfish/dto/raw-tx.dto';
+import { TransactionDto } from '../dto/transaction.dto';
+import { Transaction } from '../types/transaction';
+import { TxCheck } from '../util/tx-check';
+
+@Injectable()
+export class TransactionService {
+  private readonly transactions: Map<string, Transaction>;
+
+  constructor() {
+    this.transactions = new Map<string, Transaction>();
+  }
+
+  getOpen(): TransactionDto[] {
+    return this.select((tx: TransactionDto) => TxCheck.isOpen(tx));
+  }
+
+  getVerified(): TransactionDto[] {
+    return this.select((tx: TransactionDto) => TxCheck.isVerified(tx));
+  }
+
+  verify(id: string, signature: string) {
+    const tx = this.transactions.get(id);
+    if (!tx) throw new NotFoundException('Transaction not found');
+    tx.verifierSignature = signature;
+    this.transactions.set(tx.id, tx);
```

Comment:
Is this line required?

Reply:
ah thanks, i have written a test case, but forgot to check if this line is needed and no it is not